### PR TITLE
Basic iocard emulation

### PR DESCRIPTION
**Pull request by finomen**

### CODE REVIEW

File: hw/misc/iocard.c [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+};
+
+
+static uint64_t iocard_mmio_read(void *opaque, hwaddr addr, unsigned size)
+{
+    IocardState *iocard = opaque;
+    uint64_t val = ~0ULL;
+
+    switch(addr) {
+    case IOCARD_BLDDATE_OFFSET:
```
Comment (Javier-varez) [inline hw/misc/iocard.c:58]: nit: Use scoped case statements

File: hw/misc/iocard.c [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        .max_access_size = 8,
+    },
+};
+
+static void pci_iocard_realize(PCIDevice *pdev, Error **errp)
+{
+    IocardState *iocard = IOCARD(pdev);
+    memory_region_init_io(&iocard->mmio, OBJECT(iocard), &iocard_mmio_ops, iocard,
+            "iocard-mmio", 0x400000);
+    pci_register_bar(pdev, 0, PCI_BASE_ADDRESS_SPACE_MEMORY, &iocard->mmio);
```
Comment (Javier-varez) [inline hw/misc/iocard.c:100]: nit: constant for the bar index

File: hw/misc/iocard.c [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+static uint64_t iocard_mmio_read(void *opaque, hwaddr addr, unsigned size)
+{
+    IocardState *iocard = opaque;
+    uint64_t val = ~0ULL;
+
+    switch(addr) {
+    case IOCARD_BLDDATE_OFFSET:
+      val = val  << 32;
+      val |= 0xB62CCCED;
```
[the remark this answers is not in the document]
Comment (finomen) [reply hw/misc/iocard.c:60]: Actual values from vantage board

File: hw/misc/iocard.c [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    MemoryRegion mmio;
+    bool wow;
+    bool maintenance;
+};
+
+
+static uint64_t iocard_mmio_read(void *opaque, hwaddr addr, unsigned size)
+{
+    IocardState *iocard = opaque;
+    uint64_t val = ~0ULL;
```
[the remark this answers is not in the document]
Comment (finomen) [reply hw/misc/iocard.c:55]: Changed max read size to 4. If fill upped 4 bytes with 0 something goes wrong and all reads except first return 0xFAFAFAFA. Iocard support only 32-bit read and write access

**Review by Javier-varez — APPROVED** [review]

File: hw/misc/iocard.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (Javier-varez) [reply hw/misc/iocard.c:58]: That's very strange behavior... There has to be some magic causing that, but I have no idea what it could be

**Pull request merged**